File: src/types/evidence.ts
import type { ConfidenceLevel, CorePhase } from "./phases";

export type EvidenceType =
  | "general"
  | "observation"
  | "quote"
  | "pain_point"
  | "jtbd"
  | "assumption"
  | "hypothesis"
  | "insight";

export const EVIDENCE_TYPE_LABELS: Record<EvidenceType, string> = {
  general: "General",
  observation: "Observation",
  quote: "Quote",
  pain_point: "Pain Point",
  jtbd: "Job-to-be-Done",
  assumption: "Assumption",
  hypothesis: "Hypothesis",
  insight: "Insight",
};

export interface Evidence {
  id: string;
  discovery_id: string;
  phase: CorePhase;
  content: string;
  source: string;
  confidence: ConfidenceLevel;
  evidence_type: EvidenceType;
  tags: string[];
  created_at: string;
}

export interface Assumption {
  id: string;
  text: string;
  risk: "high" | "medium" | "low";
  status: "untested" | "validated" | "invalidated";
  certainty?: "high" | "medium" | "low" | "unknown";
  evidence?: string;
  validation_method?: string;
  owner?: string;
  impact_if_wrong?: string;
}

File: backend/app/models/core/evidence.py
from datetime import datetime

from pydantic import BaseModel, Field

from ._helpers import _utcnow
from .enums import ConfidenceLevel, CorePhase, EvidenceType


class Evidence(BaseModel):
    id: str = ""
    project_id: str = ""
    discovery_id: str = ""
    phase: CorePhase
    content: str
    source: str = ""
    confidence: ConfidenceLevel = ConfidenceLevel.UNKNOWN
    evidence_type: EvidenceType = EvidenceType.GENERAL
    tags: list[str] = Field(default_factory=list)
    created_by: str = ""
    updated_by: str = ""
    created_at: datetime = Field(default_factory=_utcnow)


class EvidenceUpdate(BaseModel):
    """Typed update payload for evidence items."""

    content: str | None = None
    source: str | None = None
    confidence: ConfidenceLevel | None = None
    evidence_type: EvidenceType | None = None
    tags: list[str] | None = None
